src/lints/struct_with_no_pub_fields_changed.ron

```diff
@@ -0,0 +1,77 @@
+Changing such a struct to make it an enum or union will break this pattern match, as their syntax is different.
+
+More info: https://github.com/obi1kenobi/cargo-semver-checks/issues/954
+"#
+    ),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Struct {
+                        struct_typename: __typename @tag @output
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        struct_type @output
+
+                        # Ensure the struct does not have pub fields to prevent overlap with struct_with_pub_fields_changed
+                        field @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                            visibility_limit @filter(op: "=", value: ["$public"])
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+                        }
+
+                        # Ensure the struct does have non-pub fields to prevent overlap with constructible_struct_changed_type
+                        field @fold @transform(op: "count") @filter(op: ">", value: ["$zero"]) {
+                            visibility_limit @filter(op: "!=", value: ["$public"])
+                        }
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on ImplOwner {
+                        current_typename: __typename @filter(op: "!=", value: ["%struct_typename"])
+                                                     @output
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        name @output
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "true": true,
+        "zero": 0,
+    },
+    error_message: "A struct without pub fields became an enum or union, breaking pattern matching.",
+    per_result_error_template: Some("struct {{join \"::\" path}} became {{lowercase current_typename}} in file {{span_filename}}:{{span_begin_line}}"),
+    witness: (
+        hint_template: r#"matches!(value, {{join "::" path}}{...});"#,
+        witness_template: r#"matches!(value, {{join "::" path}}::Variant);"#,
```

Comment:
Bit unsure if I should fill out both fields here and if their usage (what should I fill into each of the both fields) is.
`hint_template` is clear enough, but I don't get what `witness_template` is.

Need to review what the original PR intendeded to put here.
Since no other lint uses `witness_query` or `witness_template` this is a bit difficult ^^

Likely including a sentence what `witness_template` does in relation to `hint_template` would be helpfull in the contribution guide.
Likely this is [this todo](https://github.com/obi1kenobi/cargo-semver-checks/blob/main/CONTRIBUTING.md#full-witness-templates).

Reply:
`witness_template` is not implemented yet, so you can leave it out for now.  it will be a full buildable program that we can use to prove that downstream could break with this change. 

```suggestion
        hint_template: r#"matches!(value, {{join "::" path}}{...});"#,
```